src/server/report/app.js

```diff
@@ -141,24 +147,28 @@ class App extends LitElement {
 		`;
 	}
 	_goHome() {
-		page(this._root);
+		this._updateSearchParams({ file: undefined, test: undefined });
 	}
 	_handleFilterBrowserChange(e) {
-		const index = this._filterBrowsers.indexOf(e.target.value);
-		if (!e.target.checked && index > -1) {
-			this._filterBrowsers.splice(index, 1);
-		} else if (e.target.checked && index === -1) {
-			this._filterBrowsers.push(e.target.value);
-		}
-		this._updateFiles();
+		const browsers = data.browsers.map(b => b.name).filter(b => {
+			if (b === e.target.value) {
+				return e.target.checked;
+			} else {
+				return this._filterBrowsers.includes(b);
+			}
+		});
```

Comment:
Minor tweak here to use the original browser list to rebuild the filter so that browsers are always rendered in the same order.